airflow/providers/amazon/aws/executors/ecs/ecs_executor.py

```diff
@@ -386,14 +385,16 @@ def attempt_task_runs(self):
                     )
                     self.pending_tasks.append(ecs_task)
                 else:
-                    self.log.error(
-                        "ECS task %s has failed a maximum of %s times. Marking as failed",
+                    self.task_context_logger.error(
```

Comment:
similar comment here. just consider it everywhere. are you introducing usages of taskcontextlogger that (1) are not really necessary because the user will get enough information already or (2) just occur too frequently, so that the scheduler will now have to upload logs to s3 routinely during its loop.

because of the performance considerations, the feature should really only be used when the user would otherwise have no information about what happened.

Reply:
That's a good call. I can go through again the different scenarios and narrow them down to a few. TBH, I just went through all the `self.log.error(` thinking whatever error happens on the executor side, the user should know about it. But I guess I can apply a slightly smarter filter

Reply:
I narrowed it down to one scenario in `base_executor` and two in `ecs_executor`. Let me know what you think.